src/utils.cpp

```diff
@@ -764,5 +771,250 @@ bool executePreAction(const nlohmann::json& i_parsedConfigJson,
+
+            auto mapperObjectMap = getObjectMap(constants::pimPath,
+                                                pimInterface);
+
+            if (mapperObjectMap.empty())
+            {
+                throw std::runtime_error(
+                    "D-bus object not found for the input provided.");
+            }
+
+            l_service = mapperObjectMap.begin()->first.c_str();
+        }
+
+        std::string l_keyword{}, l_interface{};
+
+        if (const types::IpzType* l_ipzData =
+                std::get_if<types::IpzType>(&i_paramsToReadData))
+        {
+            l_interface = constants::ipzVpdInf + std::get<0>(*l_ipzData);
+            l_keyword = std::get<1>(*l_ipzData);
+        }
+
+        return (
+            utils::readDbusProperty(l_service, i_path, l_interface, l_keyword));
+    }
+}
+
+std::string getInvPath(const nlohmann::json i_jsonObj,
+                       const std::string& i_path)
+{
+    // Given any path, fetch the inventory path from json
+    if (i_path.empty())
+    {
+        throw(DataException("Invalid File path provided."));
+    }
+
+    if (!i_jsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in VPD JSON");
+    }
+
+    const nlohmann::json& listOfFrus =
+        i_jsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& itemFRUS : listOfFrus.items())
+    {
+        const auto l_hwPath = itemFRUS.key();
+        const auto l_redundantHwPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("redundantEeprom", "");
+        const auto l_invPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("inventoryPath", "");
+
+        if (i_path == l_hwPath || i_path == l_redundantHwPath ||
+            i_path == l_invPath)
+        {
+            return l_invPath;
+        }
+    }
+
+    return std::string();
+}
+
+std::string getRedundantPath(const nlohmann::json i_jsonObj,
+                             const std::string& i_path)
+{
+    // Given any path, fetch the redundant eeprom path from json
+    if (i_path.empty())
+    {
+        throw(DataException("Invalid File path provided."));
+    }
+
+    if (!i_jsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in VPD JSON");
+    }
+
+    const nlohmann::json& listOfFrus =
+        i_jsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& itemFRUS : listOfFrus.items())
+    {
+        const auto l_hwPath = itemFRUS.key();
+        const auto l_redundantHwPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("redundantEeprom", "");
+        const auto l_invPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("inventoryPath", "");
+
+        if (i_path == l_hwPath || i_path == l_redundantHwPath ||
+            i_path == l_invPath)
+        {
+            return l_redundantHwPath;
+        }
+    }
+
+    return std::string();
+}
+
+std::string getPrimaryPath(const nlohmann::json i_jsonObj,
```

Comment:
is API name should be getPrimaryHwPath ?

Reply:
done